daliuge-engine/dlg/data/drops/memory.py

```diff
@@ -165,34 +186,32 @@ def initialize(self, **kwargs):
         will use StringIO as the buffer. This can have ramifications for none-string
         encodings used further down the track.
         """
+        # TODO: We need to ignore pydata if we have a connected input port
```

Comment:
@awicenec why do we need to ignore `pydata` if we have a connected input port? The input port will overwrite the data in the buffer when it is finished, so it will be 'ignored' during the graph runtime. 

This would of course break if had type mis-matches, but I think that we should just make that a runtime error if someone produces incorrectly typed input to a typed memory drop.